### PR TITLE
migrate to null safety

### PR DESCRIPTION
**Pull request by davidp918**
Hi! I tried to migrate it but I'm not experienced with library stuff, please review!

### CODE REVIEW

File: lib/src/openid.dart [hunk 1]
```diff
@@ -40,15 +40,15 @@ class OpenId {
...
+    Uri uri = _host!.startsWith('https')
         ? Uri.https('steamcommunity.com', '/openid/login', data)
         : Uri.http('steamcommunity.com', '/openid/login', data);
     return uri;
   }
 
   /// Must be called only when mode is 'id_res' or an [OpenIdException] will be thrown.
   /// Validates the authentication and return a [Future] string with the user's steamid64.
-  Future<String> validate() async {
+  Future<String?> validate() async {
```
Comment (Hexer10) [inline lib/src/openid.dart]: This should be non-nullable: `String`

File: lib/src/openid.dart [hunk 2]
```diff
@@ -91,20 +88,20 @@ class OpenId {
...
 
-    var openIdUrl = _data['openid.claimed_id'];
+    var openIdUrl = _data['openid.claimed_id']!;
     if (!_validation_regexp.hasMatch(openIdUrl)) {
       throw OpenIdException(
           OpenIdFailReason.pattern, 'Invalid steam id pattern');
     }
 
-    return _validation_regexp.firstMatch(openIdUrl).group(1);
+    return _validation_regexp.firstMatch(openIdUrl)!.group(1);
```
Comment (Hexer10) [inline lib/src/openid.dart]: Add an `!` after .group(1) so it becomes non-null.

File: lib/src/openid.dart [hunk 3]
```diff
@@ -16,9 +16,9 @@ class OpenId {
   final RegExp _validation_regexp =
       RegExp(r'^https://steamcommunity.com/openid/id/(7[0-9]{15,25})$');
 
-  String _host;
-  String _returnUrl;
-  Map<String, String> _data;
+  String? _host;
+  String? _returnUrl;
+  late Map<String, String> _data;
```
Comment (Hexer10) [inline lib/src/openid.dart]: You should be able to make these variables `late final` (remeber to remove the custom getters afterwards and change all the usages from _<var> to <var> ie: _host to host)
``` 
late final String host;
late final String returnUrl;
late final Map<String, String> data;
``

File: lib/src/openid.dart [hunk 4]
```diff
@@ -91,20 +88,20 @@ class OpenId {
...
+    return _validation_regexp.firstMatch(openIdUrl)!.group(1);
   }
 
   /// Current [host].
-  String get host => _host;
+  String? get host => _host;
 
   /// Current [returnUrl]
-  String get returnUrl => _returnUrl;
+  String? get returnUrl => _returnUrl;
```
Comment (Hexer10) [inline lib/src/openid.dart]: Remove these two (Remember to move the doc comment in their declartion).

File: lib/src/player.dart [hunk 5]
```diff
@@ -7,10 +7,11 @@ import 'exceptions.dart';
 /// Calls the [GetPlayerSummaries] SteamAPI, and returns a Map containing the data
 /// Listed on https://developer.valvesoftware.com/wiki/Steam_Web_API#GetPlayerSummaries_.28v0002.29
 /// Throws a Generic
-Future<Map> GetPlayerSummaries(String steamid, String apikey) async {
+Future<Map<String, dynamic>> GetPlayerSummaries(
+    String? steamid, String apikey) async {
```
Comment (Hexer10) [inline lib/src/player.dart]: `steamid` should be non-nullable

**Review by Hexer10 — CHANGES_REQUESTED** [review]

**Comment by Hexer10** [comment]
@davidp918  Thanks for the PR! I just added some comments around the code, update the code and I'll merge the PR. If you don't get something feel freee to contact me

**Comment by davidp918** [comment]
@Hexer10 Thanks for the review! I've just updated and committed the code.

**Review by Hexer10 — APPROVED** [review]
Nice, thank you!

**Pull request merged**